modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Traverse.kt

```diff
@@ -5,11 +5,526 @@ import arrow.core.Id
+ *
+ * Given just `(User) -> IO<Profile>`, what should we do if we want to fetch profiles for a `List<User>`? We could try familiar combinators like map.
+ *
+ * ```kotlin
+ * fun profilesFor(users: List<User>): List<IO<Profile>> =
+ *   users.map(::userInfo)
+ * ```
+ *
+ * Note the return type `List<IO<Profile>>`. This makes sense given the type signatures, but seems unwieldy. We now have a list of asynchronous values, and to work with those values we must then use the combinators on `IO` for every single one. It would be nicer instead if we could get the aggregate result in a single `IO`, say a `IO<List<Profile>>`.
+ *
+ * ### Sequencing
+ *
+ * Similar to the latter, you may find yourself with a collection of data, each of which is already in an data type, for instance a `List<Option<A>>` or `List<IO<Profile>>`. To make this easier to work with, you want a `Option<List<A>>` or `IO<List<Profile>>`. Given `Option` and `IO` have an [Applicative] instance, we can [sequence] the list to reverse the types.
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.Option
+ * import arrow.core.extensions.list.traverse.sequence
+ * import arrow.core.extensions.option.applicative.applicative
+ * import arrow.core.none
+ * import arrow.core.some
+ *
+ * fun main() {
+ *   //sampleStart
+ *   val optionList =
+ *     listOf(1.some(), 2.some(), 3.some())
+ *       .sequence(Option.applicative())
+ *
+ *   val emptyList =
+ *     listOf(1.some(), none(), 3.some())
+ *       .sequence(Option.applicative())
+ *   //sampleEnd
+ *   println("optionList = $optionList")
+ *   println("emptyList = $emptyList")
+ * }
+ * ```
+ *
+ * [Traverse] provides [sequence] as a convenience method for `traverse(::identity)`.
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.extensions.option.applicative.applicative
+ * import arrow.core.extensions.option.applicative.map
+ * import arrow.core.fix
+ * import arrow.core.identity
+ * import arrow.core.none
+ * import arrow.core.some
+ * import arrow.core.Option
+ * import arrow.core.extensions.list.traverse.traverse
+ *
+ * fun main() {
+ *   //sampleStart
+ *   val optionList: Option<List<Int>> =
+ *     listOf(1.some(), 2.some(), 3.some())
+ *       .traverse(Option.applicative(), ::identity).fix().map { it.fix() }
+ *
+ *   val emptyList: Option<List<Int>> =
+ *     listOf(1.some(), none(), 3.some())
+ *       .traverse(Option.applicative(), ::identity).map { it.fix() }
+ *   //sampleEnd
+ *   println("optionList = $optionList")
+ *   println("emptyList = $emptyList")
+ * }
+ * ```
+ *
+ * Interestingly enough, [traverse] is a much more generalized and powerful form of [sequence] that would allow us to parse a `List<String>` or validate credentials for a `List<User>`.
```

Comment:
```suggestion
 * [traverse] is a much more generalized and powerful form of [sequence] that would allow us to parse a `List<String>` or validate credentials for a `List<User>`.
```
I'm no native english speaker, but leading sentences like this seems weird. Also begs the question: "interestingly enough" for whom? Also why "enough". Imo just drop it, it adds no value to the text.

Reply:
It's really just a stylistic choice of my writing. We will have a proper language review from fellow Arrow Maintainers soon. So I'd leave this as is once it goes through language review there will be more edits to this :)